cpp/source/deal.II/geometry.templates.h

```diff
@@ -124,6 +133,34 @@ void merge_components(
 }
 
 template <int dim>
+unsigned int Geometry<dim>::compute_cell_weight(
+    typename dealii::Triangulation<dim, dim>::cell_iterator const &cell,
+    std::array<unsigned int, 4> const &weights)
+{
+  // Cells in the anode of the cathode have to deal with two physics instead of
+  // only one in the collectors and the separator. Each cell starts with a
+  // default weight of 1000. This function returns the extra weight on some of
+  // the cells.
+  dealii::types::material_id material = cell->material_id();
+  std::vector<dealii::types::material_id> &anode = (*_materials)["anode"];
+  std::vector<dealii::types::material_id> &cathode = (*_materials)["cathode"];
+  std::vector<dealii::types::material_id> &separator =
+      (*_materials)["separator"];
+  std::vector<dealii::types::material_id> &collector =
+      (*_materials)["collector"];
+  if (std::find(anode.begin(), anode.end(), material) != anode.end())
+    return weights[internal::WEIGHT_TYPE::anode];
+  if (std::find(cathode.begin(), cathode.end(), material) != cathode.end())
+    return weights[internal::WEIGHT_TYPE::cathode];
+  if (std::find(separator.begin(), separator.end(), material) !=
+      separator.end())
+    return weights[internal::WEIGHT_TYPE::separator];
+  if (std::find(collector.begin(), collector.end(), material) !=
+      collector.end())
+    return weights[internal::WEIGHT_TYPE::collector];
+}
```

Comment:
why do you even bother with an enum?
you could have something like
```
dealii::types::material_id const material_id = cell->material_id();
for (auto const material_name : {"anode", "cathode", "separator", "collector"})
{
  std::vector<dealii::types::material_id> const ids = (*_materials)[material_name];
  if (std::find(ids.begin(), ids.end(), material_id) != ids.end())
     return weights[material_name];
}
throw std::runtime_error("oh no, your material id is not listed..")
```
where ``weights`` is something like a ``std::map<std::string, unsigned int>``